### ISSUE TITLE
Redirect Principles.green to new Training website 

### ISSUE THREAD

**Issue by holanita** [Green-Software-Foundation/training#109]
Why: the principles.green site still gets a decent amount of traffic, but the content is outdated
What: we should redirect traffic to the new page

Redirect from: https://principles.green/
Redirect to: https://learn.greensoftware.foundation/practitioner/introduction
To be discussed/confirmed if we should redirect pages one by one or you'll just redirect everyone to the main page regardless of which old principles.green subpage they land on?

@osamajandali can you get started on this one or do you need further info?

**Comment by holanita** [comment]
@zanete hi. this one is not linked to any sprint yet. can you help schedule this in depending on Osama's availability?

**Comment by zanete** [comment]
@holanita will meet with him today to discuss! :)

**Comment by zanete** [comment]
this kind of stuff can only be done by @jawache

**Comment by holanita** [comment]
@zanete ok noted. did you guys define an ETA for this? should I follow up with Asim?

**Comment by zanete** [comment]
@holanita Asim will do this Friday morning after he's updated the website as per #111

**Comment by jawache** [comment]
@holanita let's pause on this one till after we've also moved to greensoftware.org, otherwise there are too many changes in one go and it makes it harder to ensure things are working

**Comment by holanita** [comment]
OK, I've updated the statuses accordingly.

**Comment by zanete** [comment]
@jawache it's been a week now, do you think it's now a better time to implement?

**Comment by holanita** [comment]
As discussed in today's team call, @jawache will do this today / tomorrow.

**Comment by jawache** [comment]
If someone can do me a favor and try to map the pages from principles.green to pages on our website? When you redirect old stuff to new stuff it's far better if you can direct a specific page to a specific page. I think some of the principles pages map to patterns instead perhaps, but if someone can get me those that's where i'm stuck and struggling for time to figure out.

**Comment by zanete** [comment]
@holanita who is best placed to do this? is this something I can do?

**Comment by holanita** [comment]
@zanete I've started adding the urls to [this doc](https://docs.google.com/spreadsheets/d/1tCAiXIY5lE-y4xWuHn_FX6imqnHg2v0WtkQEX5Ioqwo/edit#gid=0). Feel free to add more to column A if you spot something that I've missed. I can do the rest and will ask Sarah if there is something I can't match on the new website.

**Comment by holanita** [comment]
@jawache you can use [my doc/sheet](https://docs.google.com/spreadsheets/d/1tCAiXIY5lE-y4xWuHn_FX6imqnHg2v0WtkQEX5Ioqwo/edit#gid=0) to match the urls. some of the yellow and grey ones I don't know much about.

**Comment by jawache** [comment]
This is for me, principles.green is my personal site 🙂 thanks Anita!

**Comment by holanita** [comment]
@jawache I just figured recently that the old site might still get traffic bc it is reference in our menu. will create a ticket for Osama to replace the link with the new one
<img width="1336" alt="Screenshot 2023-01-16 at 19 39 20" src="https://user-images.githubusercontent.com/100846607/212746510-9da1c4c7-7f3c-4e68-9ad7-bd5301d6990a.png">

**Comment by zanete** [comment]
@osamajandali 
- [ ] move the domain to the GSF cloudflare account (from Asim personal)  - potentially
- [ ] Redirect using cliend-side redirects (https://github.com/jawache/principles-green) based on the mappings here https://docs.google.com/spreadsheets/d/1tCAiXIY5lE-y4xWuHn_FX6imqnHg2v0WtkQEX5Ioqwo/edit#gid=0

**Comment by zanete** [comment]
capturing state of issue on weekly web call: This is sitll in progress, researched various ways and concluded client side is the best

**Comment by zanete** [comment]
capturing state of issue on weekly web call: 
Discussions with @jawache have been happening on WhatsApp about this
* principles.green is now hosted on the GSF cloudflare account
* before removing the old website, need to double check with @jawache if the data is needed (e.g. analytics) or can it be removed.
* redirects are still to be implemented in code.

**Comment by osamajandali** [comment]
I finished this one but this is an issue when building, something related to not have the authority to certain tasks
https://github.com/jawache/principles-green/actions/runs/4248091303/jobs/7386857285
@jawache could you please check that as you are the owner of the repo

**Comment by zanete** [comment]
Capturing the status during the weekly call:
- @jawache needs to carry out the build as Osama doesn't have permissions or something isn't right with settings

**Comment by zanete** [comment]
Capturing the status during the weekly call:
- Still awaiting on @jawache to complete this

**Comment by zanete** [comment]
Capturing the status during the weekly call:
- Still awaiting on @jawache to complete this

**Comment by osamajandali** [comment]
@jawache Could you try manual deploy from your end? just to see if this is the problem or not. And if there is more level of permission I can take please give it to me.

**Comment by jawache** [comment]
Ahh ok, Ill try it on my old laptop

**Comment by zanete** [comment]
Capturing the status during the weekly call:
* @jawache let us know how that went so we can close

**Comment by jawache** [comment]
Tried in several places, old laptop is fine. I can see it was erroring out on github pages with some strange error with deploy.

The github actions config was using the v2 of something, the latest is v3 thought it might be that - rebuilding. https://github.com/jawache/principles-green/commit/fe33b2ad86cb486cacf0650b633eb803744eff69

**Comment by zanete** [comment]
Capturing status in weekly call:
- @osamajandali will look at it this week

**Comment by zanete** [comment]
Capturing status in weekly call:
- Investigation is still ongoing, no solution yet
- will give it another try this week and hope that a solution can be found

**Comment by zanete** [comment]
Capturing status in weekly call:
- has been on hold due to SOGS, but will carry on this week

**Comment by osamajandali** [comment]
@jawache @zanete I fixed the version problem, now there is another issue coming from the tokens. @jawache Can you please give me full access to repo? so I can investigate further

**Comment by jawache** [comment]
I've checked and there is no way to give you any more access, "collaborator" level seems to be the highest?

![image](https://github.com/Green-Software-Foundation/learn/assets/897523/b11faf10-48b4-4634-bf10-c713eab85703)

**Comment by jawache** [comment]
@osamajandali given that this is going to be a complete one-off last-ever change to this repo, can we just edit the gh-pages branch and push the changes manually to there? https://github.com/jawache/principles-green/tree/gh-pages

**Comment by zanete** [comment]
@osamajandali given what Asim says above, have unblocked and put back in progress.

**Comment by osamajandali** [comment]
Finally !
I was expecting that we need to proper build it so we may do more edits in the future but because this repo will not be touch @jawache solution works perfectly.
Now if you visit https://principles.green/ or any of it's other pages, you will be redirected to learn website.
